File: app/src/main/java/dev/ragnarok/fenrir/api/model/server/VkApiWallUploadServer.java
package dev.ragnarok.fenrir.api.model.server;

import android.os.Parcel;
import android.os.Parcelable;

import androidx.annotation.Nullable;

public class VKApiWallUploadServer implements Parcelable, UploadServer {

    public static final Creator<VKApiWallUploadServer> CREATOR = new Creator<VKApiWallUploadServer>() {
        @Override
        public VKApiWallUploadServer createFromParcel(Parcel in) {
            return new VKApiWallUploadServer(in);
        }

        @Override
        public VKApiWallUploadServer[] newArray(int size) {
            return new VKApiWallUploadServer[size];
        }
    };
    /**
     * адрес для загрузки фотографий
     */
    @Nullable
    public String upload_url;
    /**
     * идентификатор альбома, в который будет загружена фотография
     */
    public int album_id;
    /**
     * идентификатор пользователя, от чьего имени будет загружено фото
     */
    public int user_id;

    public VKApiWallUploadServer() {

    }

    protected VKApiWallUploadServer(Parcel in) {
        upload_url = in.readString();
        album_id = in.readInt();
        user_id = in.readInt();
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeString(upload_url);
        dest.writeInt(album_id);
        dest.writeInt(user_id);
    }

    @Nullable
    @Override
    public String getUrl() {
        return upload_url;
    }
}

File: app_fenrir/src/main/kotlin/dev/ragnarok/fenrir/api/model/server/VKApiWallUploadServer.kt
package dev.ragnarok.fenrir.api.model.server

import android.os.Parcel
import android.os.Parcelable
import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
class VKApiWallUploadServer : Parcelable, UploadServer {
    /**
     * адрес для загрузки фотографий
     */
    @SerialName("upload_url")
    override var url: String? = null

    /**
     * идентификатор альбома, в который будет загружена фотография
     */
    var album_id = 0

    /**
     * идентификатор пользователя, от чьего имени будет загружено фото
     */
    var user_id = 0

    @Suppress("UNUSED")
    constructor()
    internal constructor(parcel: Parcel) {
        url = parcel.readString()
        album_id = parcel.readInt()
        user_id = parcel.readInt()
    }

    override fun describeContents(): Int {
        return 0
    }

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeString(url)
        dest.writeInt(album_id)
        dest.writeInt(user_id)
    }

    companion object CREATOR : Parcelable.Creator<VKApiWallUploadServer> {
        override fun createFromParcel(parcel: Parcel): VKApiWallUploadServer {
            return VKApiWallUploadServer(parcel)
        }

        override fun newArray(size: Int): Array<VKApiWallUploadServer?> {
            return arrayOfNulls(size)
        }
    }
}